umh-core/pkg/fsm/dataflowcomponent/actions.go

```diff
@@ -16,46 +16,234 @@ package dataflowcomponent
+		return fmt.Errorf("failed to start DataflowComponent service %s: %w", d.baseFSMInstance.GetID(), err)
+	}
+
+	d.baseFSMInstance.GetLogger().Debugf("DataflowComponent service %s start command executed", d.baseFSMInstance.GetID())
 	return nil
 }
 
-// StopInstance is called when the container monitoring should be disabled.
-// Currently this is a no-op as the monitoring service runs independently.
+// StopInstance attempts to stop the DataflowComponent by setting the desired state to stopped for the given instance
 func (d *DataflowComponentInstance) StopInstance(ctx context.Context, filesystemService filesystem.Service) error {
-	d.baseFSMInstance.GetLogger().Infof("Disabling monitoring for %s (no-op)", d.baseFSMInstance.GetID())
+	d.baseFSMInstance.GetLogger().Debugf("Starting Action: Stopping DataflowComponent service %s ...", d.baseFSMInstance.GetID())
+
+	// Set the desired state to stopped for the given instance
+	err := d.service.StopDataFlowComponent(ctx, filesystemService, d.baseFSMInstance.GetID())
+	if err != nil {
+		// if the service is not there yet but we attempt to stop it, we need to throw an error
+		return fmt.Errorf("failed to stop DataflowComponent service %s: %w", d.baseFSMInstance.GetID(), err)
+	}
+
+	d.baseFSMInstance.GetLogger().Debugf("DataflowComponent service %s stop command executed", d.baseFSMInstance.GetID())
 	return nil
 }
 
-// UpdateObservedStateOfInstance is called when the FSM transitions to updating.
-// For container monitoring, this is a no-op as we don't need to update any resources.
+// getServiceStatus gets the status of the DataflowComponent service
+// its main purpose is to handle the edge cases where the service is not yet created or not yet running
+func (d *DataflowComponentInstance) getServiceStatus(ctx context.Context, filesystemService filesystem.Service, tick uint64) (dataflowcomponentservice.ServiceInfo, error) {
+	info, err := d.service.Status(ctx, filesystemService, d.baseFSMInstance.GetID(), tick)
+	if err != nil {
+		// If there's an error getting the service status, we need to distinguish between cases
+
+		if errors.Is(err, dataflowcomponentservice.ErrServiceNotExist) {
+			// If the service is being created, we don't want to count this as an error
+			// The instance is likely in Creating or ToBeCreated state, so service doesn't exist yet
+			// This will be handled in the reconcileStateTransition where the service gets created
+			if d.baseFSMInstance.GetCurrentFSMState() == internalfsm.LifecycleStateCreating ||
+				d.baseFSMInstance.GetCurrentFSMState() == internalfsm.LifecycleStateToBeCreated {
+				return dataflowcomponentservice.ServiceInfo{}, dataflowcomponentservice.ErrServiceNotExist
+			}
+
+			// Log the warning but don't treat it as a fatal error
+			d.baseFSMInstance.GetLogger().Debugf("Service not found, will be created during reconciliation")
+			return dataflowcomponentservice.ServiceInfo{}, nil
```

Comment:
In benthos for example we return partial info:

			infoWithFailedHealthChecks := info
			infoWithFailedHealthChecks.BenthosStatus.HealthCheck.IsLive = false
			infoWithFailedHealthChecks.BenthosStatus.HealthCheck.IsReady = false
			// Return ServiceInfo with health checks failed but preserve S6FSMState if available
			return infoWithFailedHealthChecks, nil
			
might make sense here too

Reply:
Returning an empty `ServiceInfo` should already have all the boolean values set to false. Not sure if I understood your comment properly. @Scarjit

Reply:
Done